diff --git a/source/blender/editors/interface/interface_handlers.cc b/source/blender/editors/interface/interface_handlers.cc
index cc549e7f7ad..0733b165377 100644
--- a/source/blender/editors/interface/interface_handlers.cc
+++ b/source/blender/editors/interface/interface_handlers.cc
@@ -3720,10 +3720,12 @@ static void button_edit_unit_hint_refresh(bContext *C, Button *but, HandleButton
     }
 
     /* If the expression we're entering is not valid, don't show the hint. */
+#ifdef WITH_PYTHON
     if (!BPY_string_compile_check(data->text_edit.edit_string)) {
       data->text_edit_unit_hint.clear();
       return;
     }
+#endif
 
     const void *usys;
     int len;
@@ -3746,10 +3748,12 @@ static void button_edit_unit_hint_refresh(bContext *C, Button *but, HandleButton
     }
 
     /* If the expression we're entering is not valid, don't show the hint. */
+#ifdef WITH_PYTHON
     if (!BPY_string_compile_check(data->text_edit.edit_string)) {
       data->text_edit_unit_hint.clear();
       return;
     }
+#endif
   }
 
   if (name_short.empty()) {
