diff --git a/source/blender/gpu/webgpu/wgpu_batch.cc b/source/blender/gpu/webgpu/wgpu_batch.cc
--- a/source/blender/gpu/webgpu/wgpu_batch.cc
+++ b/source/blender/gpu/webgpu/wgpu_batch.cc
@@ -120,6 +120,9 @@ static wgpu::ComputePipeline triangle_fan_index_pipeline(const wgpu::Device &dev
   wgpu::ShaderModuleDescriptor shader_descriptor = {};
   shader_descriptor.nextInChain = &wgsl;
   wgpu::ShaderModule module = device.CreateShaderModule(&shader_descriptor);
+  if (module == nullptr) {
+    return nullptr;
+  }
   wgpu::ComputePipelineDescriptor pipeline_descriptor = {};
   pipeline_descriptor.compute.module = module;
   pipeline_descriptor.compute.entryPoint = "main";
@@ -198,10 +201,10 @@ static bool expand_indexed_triangle_fan(WGPUContext &ctx,
-  wgpu::CommandEncoder encoder = ctx.device_get().CreateCommandEncoder();
-  wgpu::ComputePassEncoder pass = encoder.BeginComputePass();
-  pass.SetPipeline(pipeline);
-  pass.SetBindGroup(0, group);
-  pass.DispatchWorkgroups(1);
-  pass.End();
-  wgpu::CommandBuffer command_buffer = encoder.Finish();
-  ctx.queue_get().Submit(1, &command_buffer);
-  return true;
+  return webgpu::command_pass_encode_submit_if_valid(
+      ctx.device_get(),
+      ctx.queue_get(),
+      [](auto &encoder) { return encoder.BeginComputePass(); },
+      [&](auto &pass) {
+        pass.SetPipeline(pipeline);
+        pass.SetBindGroup(0, group);
+        pass.DispatchWorkgroups(1);
+      });
 }
