diff --git a/CMakeLists.txt b/CMakeLists.txt
index 216dbab8b2d..12925be37ce 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -535,7 +535,7 @@ to make Blender more portable across distributions"
 endif()
 
 # Misc
-if(WIN32 OR APPLE OR ((UNIX AND (NOT HAIKU)) AND WITH_GHOST_WAYLAND))
+if(WIN32 OR APPLE OR WITH_GHOST_WEB OR ((UNIX AND (NOT HAIKU)) AND WITH_GHOST_WAYLAND))
   option(WITH_INPUT_IME "Enable Input Method Editor (IME) for complex Asian character input" ON)
 else()
   set(WITH_INPUT_IME OFF)
