diff --git a/source/blender/windowmanager/intern/wm_draw.cc b/source/blender/windowmanager/intern/wm_draw.cc
--- a/source/blender/windowmanager/intern/wm_draw.cc
+++ b/source/blender/windowmanager/intern/wm_draw.cc
@@ -1814,7 +1814,8 @@ void wm_draw_update(bContext *C)
 
   BKE_image_free_unused_gpu_textures();
 
-#ifdef WITH_METAL_BACKEND
+#if defined(WITH_METAL_BACKEND) || \
+    (defined(WITH_WEBGPU_BACKEND) && defined(__EMSCRIPTEN__))
   /* Reset drawable to ensure GPU context activation happens at least once per frame if only a
    * single context exists. This is required to ensure the default framebuffer is updated
    * to be the latest backbuffer. */
